spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/repository/query/SpannerStatementQueryExecutor.java

```diff
@@ -361,7 +361,8 @@ public static Statement buildStatementFromSqlWithArgs(String sql, List<String> t
 	private static void bindParameter(ValueBinder<Statement.Builder> bind,
 			Function<Object, Struct> paramStructConvertFunc, SpannerCustomConverter spannerCustomConverter,
 			Object originalParam, Parameter paramMetadata) {
-		if (ConversionUtils.isIterableNonByteArrayType(originalParam.getClass())) {
+		Class propType = originalParam != null ? originalParam.getClass() : paramMetadata.getType();
```

Comment:
Can you add a comment here about how you need to know the type of the parameter (even if you bind `null`) which is why the type information is relevant for null?